DetectorDescription/DDCMS/interface/DDVolumeProcessor.h

```diff
@@ -0,0 +1,91 @@
+#ifndef DETECTOR_DESCRIPTION_DD_VOLUME_PROCESSOR_H
+#define DETECTOR_DESCRIPTION_DD_VOLUME_PROCESSOR_H
+
+#include "DD4hep/VolumeProcessor.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DetectorDescription/DDCMS/interface/DDSpecParRegistry.h"
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace cms {
+  
+  class DDVolumeProcessor : public dd4hep::PlacedVolumeProcessor {
+  public:
+
+    using Volume = dd4hep::Volume;
+    using PlacedVolume = dd4hep::PlacedVolume;
+    using PlacedVolumeProcessor = dd4hep::PlacedVolumeProcessor;
+    
+    DDVolumeProcessor() = default;
+    ~DDVolumeProcessor() override = default;
+
+    std::string_view stripNamespace(std::string_view v) {
+      auto first = v.find_first_of(":");
+      v.remove_prefix(std::min(first+1, v.size()));
+      return v;
+    }
+    
+    std::string_view stripCopyNo(std::string_view v) {
+      auto found = v.find_last_of("_");
+      if(found != v.npos) {
+	v.remove_suffix(v.size() - found);
+      }
+      return v;
+    }
+    
+    std::vector<std::string_view> split(std::string_view str, const char* delims) {
```

Comment:
Should this function return a vector? Wouldn't it be faster to pass a reference to a vector into the function to avoid the copying?

Reply:
Also, how widely used is the `cms` namespace? A simple name like `split` seems like it could be used elsewhere in a big namespace.

Reply:
nope, no need to change the syntax. The compiler will take care of it.